src/redprl/refiner.fun

```diff
@@ -91,6 +91,21 @@ struct
           raise Fail "Did not match"
       end
 
+    fun FromLESS z _ jdg =
+      let
+        val _ = RedPrlLog.trace "Synth.FromLESS"
+        val (I, H) >> CJ.SYNTH tm = jdg
+        val Syn.INT = Syn.out tm
+        val CJ.LESS (a, b) = lookupHyp H z
+      in
+        if Abt.eq (a, tm) orelse Abt.eq (b, tm) then
+          T.empty #> (I, H, Syn.into Syn.INT)
+        else
+          raise Fail "Did not match"
+      end
+
+    fun FromWfHyp z = FromEQ z orelse_ FromLESS z
```

Comment:
Oh, I thought you meant the well-formedness of having EQ, not that EQ itself being the definition of well-formedness. Can this be viewed as the well-formedness of LESS? I am thinking whether we could figure out more if not restricted to well-formedness.